### PR TITLE
Finish constexpr for Point.h

### PR DESCRIPTION
**Pull request by rsjaffe**
My apologies. Missed some functions that can be made constexpr with first pull.

### CODE REVIEW

File: modules/juce_graphics/geometry/juce_Point.h [hunk 1]
```diff
@@ -25,6 +25,10 @@
 */
 
 #pragma once
+#if defined(_MSC_VER) && _MSC_VER < 1910
+#pragma push_macro("constexpr")
+#define constexpr /*constexpr*/
```
Comment (jrlanglois) [inline modules/juce_graphics/geometry/juce_Point.h]: I was under the impression that VS2013 has partial support for `constexpr`, so this code should be throwing errors at worst, warnings at best, no?

File: modules/juce_graphics/geometry/juce_Point.h [hunk 2]
```diff
@@ -41,32 +45,32 @@ class Point
 {
 public:
     /** Creates a point at the origin */
-    Point() noexcept : x(), y() {}
+    constexpr Point() noexcept : x(), y() {}
```
Comment (jrlanglois) [inline modules/juce_graphics/geometry/juce_Point.h]: The JUCE team already added a series of changes like this on `develop`: https://github.com/WeAreROLI/JUCE/commit/4a2a5294a7bca077bedf9f68e4774075ac8843f8

Comment (rsjaffe) [inline modules/juce_graphics/geometry/juce_Point.h]: The macro erases constexpr keyword if using VS2013 or 2015. That's why it's there. Microsoft uses the macro themselves for this purpose.

**Comment by rsjaffe** [comment]
Resolved conflicts with the way Juce is handling constexpr using a macro.

**Comment by julianstorer** [comment]
Thanks! I'll sort this out and push something shortly.

**Pull request closed without merging**